app/models/user.rb

```diff
@@ -145,6 +145,7 @@ class User < ApplicationRecord
   validate  :conditionally_validate_summary
   validate  :validate_mastodon_url
   validate  :validate_feed_url, if: :feed_url_changed?
+  validate  :non_banished_username, :username_changed?
```

Comment:
In the end I decided not to merge this with `unique_including_orgs_and_podcasts` into a `validate_username` method,  but I don't feel strongly either way, so if people would prefer a single method I'll change it.